### DISCUSSION TITLE
Give a short summary of changes on ghpc deploy/destroy

### DISCUSSION THREAD

**Discussion by yaroslavvb** [GoogleCloudPlatform/cluster-toolkit#1570] [Ideas and feature requests]
Right now ghpc output is too verbose. Destroying a cluster like this [yaml](https://www.dropbox.com/scl/fi/7pv9kzldjfi3jf7z092xu/dcluster-jun7.yaml?rlkey=jufmxzlyyye8odeuz23me4xuw&dl=0) prints >40 pages of changes (my tmux ran out of buffer at 40 pages).

Ideally there's be a way to see what exactly is being destroyed that fits on a couple of screens

**Comment by nick-stroud** [comment]
Thank you for your bug report. I have investigated the issue and determined that it is not a bug, but rather a feature request. 

I am moving this to the [ideas-and-feature-request discussion](https://github.com/GoogleCloudPlatform/hpc-toolkit/discussions/categories/ideas-and-feature-requests) where feature requests are collected.

One possible workaround would be to use a grep on terraform plan to summarize from a high level what is changing. 

```
./ghpc create <blueprint> -w
terraform -chdir=<deployment>/<group> init
terraform -chdir=<deployment>/<group> plan | grep '# module'
# or alternatively for destroying
terraform -chdir=<deployment>/<group> plan -destroy | grep '# module'
```

This will reduce the output to 1 line for each resource changed. Example output for a random change I happen to be executing:
```
  # module.batch-job-base.local_file.job_template will be created
  # module.batch-job-batch.local_file.job_template will be created
  # module.batch-job-hpc.local_file.job_template will be created
  # module.batch-job-ubuntu.local_file.job_template will be created
  # module.batch-login.google_compute_instance_from_template.batch_login will be destroyed
  # module.wait2.null_resource.wait_for_startup will be destroyed
  # module.batch-login.module.login_startup_script.google_storage_bucket.configs_bucket[0] will be destroyed
  # module.batch-login.module.login_startup_script.google_storage_bucket_object.scripts["0-install_ddn_lustre_client.sh"] will be destroyed
  # module.batch-login.module.login_startup_script.google_storage_bucket_object.scripts["0-mount-ddn-lustre.sh"] will be destroyed
  # module.batch-login.module.login_startup_script.google_storage_bucket_object.scripts["1-install_ddn_lustre_client.sh"] will be destroyed
  # module.batch-login.module.login_startup_script.google_storage_bucket_object.scripts["1-mount-ddn-lustre.sh"] will be destroyed
  # module.batch-login.module.login_startup_script.google_storage_bucket_object.scripts["README.md"] will be destroyed
  # module.batch-login.module.login_startup_script.google_storage_bucket_object.scripts["cloud-batch-cent-base.json"] will be destroyed
  # module.batch-login.module.login_startup_script.google_storage_bucket_object.scripts["cloud-batch-cent-hpc.json"] will be destroyed
  # module.batch-login.module.login_startup_script.google_storage_bucket_object.scripts["passed_startup_script.sh"] will be destroyed
  # module.batch-login.module.login_startup_script.random_id.resource_name_suffix will be destroyed
```

**Reply by issacg** [reply]
I like the availability of the full output, but agree that a terse option would be good.  Perhaps an addition option to d,a,s,c to show a terse summary for practitioners who use hpc-toolkit but don't actually know terraform (which I thought was part of the point of the toolkit? :))

However, for the long outputs when I touch something sensitive, I want to diff the whole thing.  My workaround is to manually run terraform plan with `less` or another pager (and then enjoy the colors which also help me spot potential problems (such as create/destroy instead of modify)
